apps/prairielearn/src/question-servers/calculation-subprocess.js

```diff
@@ -76,7 +87,7 @@ module.exports.generate = (question, course, variant_seed, callback) => {
 module.exports.grade = (submission, variant, question, question_course, callback) => {
   callFunction('grade', question_course, question, { submission, variant }).then(
     ({ data, courseIssues }) => callback(null, courseIssues, data),
-    (err) => console.log(err),
+    (err) => callback(err),
```

Comment:
🤦 this is what caused the 504s in production - the `callback` was never invoked, so request processing halted here.

Reply:
I really hate callbacks!